src/services/download/mod.rs

```diff
@@ -34,26 +34,21 @@ pub use crate::types::ObjectId;
 /// HTTP User-Agent string to use.
 const USER_AGENT: &str = concat!("symbolicator/", env!("CARGO_PKG_VERSION"));
 
-#[derive(Debug, Fail, Clone)]
-pub enum DownloadErrorKind {
-    #[fail(display = "failed to download")]
-    Io,
-    #[fail(display = "bad file destination")]
-    BadDestination,
-    #[fail(display = "failed writing the downloaded file")]
-    Write,
-    #[fail(display = "download was cancelled")]
+#[derive(Debug, Error)]
+pub enum DownloadError {
+    // TODO: this is currently an opaque `failure::Error`, as we pack all kinds of errors into this
+    #[error("failed to download")]
+    Io(failure::Error),
+    #[error("bad file destination")]
+    BadDestination(std::io::Error),
```

Comment:
You should probably annotate all these variants like `BadDestination(#[source] std::io::Error)` should you not?